### PR TITLE
create radio and switch components

### PR DESCRIPTION
**Pull request by tilucasoli**

### CODE REVIEW

File: lib/components/radio/radio.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  @override
+  Widget build(BuildContext context) {
+    var variant = active ? RadioState.active : RadioState.inactive;
+
+    final style =
+        RadioStyles.defaults().merge(_customStyle).selectVariants([variant]);
+
+    return PressableBox(
+      onPressed:
+          onChanged == null || disabled ? null : () => onChanged!(!active),
```
Comment (leoafarias) [inline lib/components/radio/radio.dart]: Let's consolidate this logic into a separate function like `handleOnChange`

File: lib/components/radio/radio.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      child: HBox(
+        style: style.row,
```
Comment (leoafarias) [inline lib/components/radio/radio.dart:37]: Are we going to keep the consistency of function syntax? Example `style.row()` instead?

File: lib/components/switch/switch.dart [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  @override
+  Widget build(BuildContext context) {
+    var variant = active ? SwitchState.active : SwitchState.inactive;
+
+    final style =
+        SwitchStyles.defaults().merge(_customStyle).selectVariants([variant]);
+
+    return PressableBox(
+      onPressed:
+          onChanged == null || disabled ? null : () => onChanged!(!active),
```
Comment (leoafarias) [inline lib/components/switch/switch.dart]: Lets move this logic also into a separate function

**Review by leoafarias — CHANGES_REQUESTED** [review]
This is looking really good!

File: lib/components/radio/radio.dart (hunk 2 above)
Comment (tilucasoli) [reply lib/components/radio/radio.dart:37]: I've developed a new solution for Styles classes. If all styles must be functions, the work of Styles classes will become simpler. We won't need to extend StyleRecipe anymore, and overriding styles will be easier by extending the Styles classes.

```dart
class RadioStyles {
  const RadioStyles();

  Style outerContainer(List<Variant> variants) => Style(
        box.width(16),
        box.height(16),
        box.alignment.center(),
        box.borderRadius.all(10),
        box.color.transparent(),
        box.border.all(width: 1.5, color: Colors.black87),
      );

  Style innerContainer(List<Variant> variants) => Style(
        box.borderRadius.all(10),
        box.color.black87(),
        RadioState.active(
          box.width(8.5),
          box.height(8.5),
        ),
        RadioState.inactive(
          box.width(0),
          box.height(0),
        ),
      );

  Style label(List<Variant> variants) => Style(
        text.style.fontSize(16),
        text.style.bold(),
        text.style.color.black87(),
      ).applyVariants(variants);

  Style row(List<Variant> variants) => Style(
        flex.mainAxisAlignment.center(),
        flex.crossAxisAlignment.center(),
        flex.mainAxisSize.min(),
        flex.gap(6),
      );
}

class RadioStylesDefaults extends RadioStyles {
  const RadioStylesDefaults();

  @override
  Style outerContainer(List<Variant> variants) {
    return super.outerContainer(variants);
  }

  @override
  Style innerContainer(List<Variant> variants) {
    return super.innerContainer(variants);
  }

  @override
  Style label(List<Variant> variants) {
    return super.label(variants);
  }

  @override
  Style row(List<Variant> variants) {
    return super.row(variants);
  }
}
```

**Review by leoafarias — APPROVED** [review]
LGTM

**Pull request merged**